bundles/org.openhab.binding.intesis/src/main/java/org/openhab/binding/intesis/internal/handler/IntesisBoxHandler.java

```diff
@@ -0,0 +1,383 @@
+        }
+    }
+
+    @Override
+    public void dispose() {
+        final ScheduledFuture<?> pollingTask = this.pollingTask;
+
+        IntesisBoxSocketApi api = this.intesisBoxSocketApi;
+
+        if (pollingTask != null) {
+            pollingTask.cancel(true);
+            this.pollingTask = null;
+        }
+        if (api != null) {
+            api.closeConnection();
+        }
+        super.dispose();
+    }
+
+    private synchronized void polling() {
+        IntesisBoxSocketApi api = this.intesisBoxSocketApi;
+        if (api != null) {
+            if (!api.isConnected()) {
+                try {
+                    api.openConnection();
+                } catch (IOException e) {
+                    updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR, e.getMessage());
+                }
+            }
+            api.sendAlive();
+        }
+    }
+
+    @Override
+    public void handleCommand(ChannelUID channelUID, Command command) {
+        IntesisBoxSocketApi api = this.intesisBoxSocketApi;
+        if (api != null) {
+            if (!api.isConnected()) {
+                logger.trace("Sending command failed, not connected");
+                return;
+            }
+            if (command instanceof RefreshType) {
+                logger.trace("Refresh channel {}", channelUID.getId());
+                api.sendQuery(channelUID.getId());
+                return;
+            }
+        }
+        String value = "";
+        switch (channelUID.getId()) {
+            case CHANNEL_TYPE_POWER:
+                if (command instanceof OnOffType) {
+                    value = command == OnOffType.ON ? "ON" : "OFF";
+                }
+                break;
+            case CHANNEL_TYPE_TARGETTEMP:
+                if (command instanceof QuantityType) {
+                    QuantityType<?> celsiusTemperature = (QuantityType<?>) command;
+                    celsiusTemperature = celsiusTemperature.toUnit(SIUnits.CELSIUS);
+                    if (celsiusTemperature != null) {
+                        double doubleValue = celsiusTemperature.doubleValue();
+                        logger.trace("targetTemp double value = {}", doubleValue);
+                        doubleValue = Math.max(minTemp, Math.min(maxTemp, doubleValue));
+                        value = String.format("%.0f", doubleValue * 10);
+                        logger.trace("targetTemp raw string = {}", value);
+                    }
+                }
+                break;
+            case CHANNEL_TYPE_MODE:
+            case CHANNEL_TYPE_FANSPEED:
+            case CHANNEL_TYPE_VANESUD:
+            case CHANNEL_TYPE_VANESLR:
+                value = command.toString();
+                break;
+        }
+        if (!value.isEmpty()) {
+            String function = IntesisBoxFunctionEnum.valueOf(channelUID.getId()).getFunction();
```

Comment:
If you are only using this enum for one-directional lookups, it would be better to define the `function` as part of the switch statement above. That way you can remove the use of the enum entirely.

Reply:
Done